### DISCUSSION TITLE
`map[key]` should always return error to handle

### DISCUSSION THREAD

**Discussion by ArtemkaKun** [vlang/v#18679] [Ideas]
### Describe the feature

Right now, `map[<wrong_key>]` will return a default value for the map value type. This provides confusion (see screenshots beloved) and leads to hidden unexpected behavior.

### Use Case

![image](https://github.com/vlang/v/assets/36485221/6b5b742a-3c97-45a2-9403-f0a98020de14)
![image](https://github.com/vlang/v/assets/36485221/afec1658-8529-4a72-96da-2f019e4ffc22)

### Proposed Solution

Force users to always handle errors when calling `map[key]` construction, like

```v
test := map[key]!

...

test := map[key] or { 100 }
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.4 a87f2d9 

### Environment details (OS name and version, etc.)

Playground

**Comment by medvednikov** [comment]
This would be a big breaking change, and I personally like having default zero values, like in Go.

@spytheman @joe-conigliaro what do you think?

**Comment by spytheman** [comment]
Imho it would be a breaking change indeed, but V will be more explicit, and less error prone with requiring the `m[k]?` , instead of retrieving/filling a default value implicitly.

**Comment by schwubldubl** [comment]
I would love to have this. It would have already saved me from an issue I had with a JSON API where omitting empty values in certain responses was not properly documented. A check like this would help prevent such errors, or at least improve the handling of them.
